src/main/java/org/elasticsearch/index/search/geo/GeoPolygonFilter.java

```diff
@@ -106,4 +136,41 @@ private static boolean pointInPolygon(GeoPoint[] points, double lat, double lon)
             return inPoly;
         }
     }
+
+    private BoundingBox getBoundingBox(GeoPoint[] points) {
+        Validate.isTrue(points.length > 0);
+        double maxLat = points[0].lat();
+        double minLat = points[0].lat();
+        double maxLon = points[0].lon();
+        double minLon = points[0].lon();
+
+        for (int i = 1; i < points.length; i++) {
+            maxLat = Math.max(maxLat, points[i].lat());
+            minLat = Math.min(minLat, points[i].lat());
+            maxLon = Math.max(maxLon, points[i].lon());
+            minLon = Math.min(minLon, points[i].lon());
+        }
+
+        GeoPoint topLeft = new GeoPoint(maxLat, minLon);
+        GeoPoint bottomRight = new GeoPoint(minLat, maxLon);
+        return new BoundingBox(topLeft, bottomRight);
+    }
+
+    private static class BoundingBox {
```

Comment:
Exactly, there's duplicate "BoundingBoxes" all over the place that need to be cleaned up. (Aggregations also has its own)  One way or another these rogue BBoxes need to be consolidated. I'm looking at brining in Apache SIS to handle this - and potentially replace much of the JTS and S4J logic (while brining in referencing system support). Looks like this is a good time to open that issue.